### ISSUE TITLE
False positive from no-unused-declaration for variable accessed from a closure

### ISSUE THREAD

**Issue by bajtos** [cartant/tslint-etc#10]
Consider the following code snippet:

```ts
    it('works', () => {
      let count = 0;
      const result = doSomething(() => count++);
      expect(result).to.equal(0);
    });

    function doSomething(factory: Function) {
      return factory();
    }
```

Using `tslint-etc@1.2.7`, I am receiving the following error:

```
Unused declarations are forbidden
```

Expected result:

No error is reported.

**Comment by cartant** [comment]
This should be fixed in 1.2.8. The rule failure was effected by the `++` operator.

**Comment by bajtos** [comment]
Thank you for a quick fix.

At the end, I decided to use a different implementation of "no unused" rule, it works better for us: https://github.com/ajafff/tslint-consistent-codestyle/blob/e5de338424cbb07acb727e9766e33a5d2eed367c/docs/no-unused.md

**Comment by cartant** [comment]
No worries. Thanks for the bug report.
